extensions/ql-vscode/src/compare-performance/remote-logs.ts

```diff
@@ -0,0 +1,568 @@
+        }
+
+        onData();
+        reportProgress(value?.length ?? 0);
+
+        await new Promise((resolve, reject) => {
+          archiveFileStream.write(value, (err) => {
+            if (err) {
+              reject(err);
+            }
+            resolve(undefined);
+          });
+        });
+      }
+
+      await new Promise((resolve, reject) => {
+        archiveFileStream.close((err) => {
+          if (err) {
+            reject(err);
+          }
+          resolve(undefined);
+        });
+      });
+    } catch (e) {
+      // Close and remove the file if an error occurs
+      archiveFileStream.close(() => {
+        void remove(archivePath);
+      });
+
+      if (e instanceof DOMException && e.name === "AbortError") {
+        const thrownError = new Error("The download timed out.");
+        thrownError.stack = e.stack;
+        throw thrownError;
+      }
+
+      throw e;
+    } finally {
+      disposeTimeout();
+    }
+
+    await this.readAndUnzip(
+      Uri.file(archivePath).toString(true),
+      unzipPath,
+      progress,
+    );
+
+    // remove archivePath eagerly since these archives can be large.
+    await remove(archivePath);
+  }
+
+  private async checkForFailingResponse(
+    response: Response,
+    errorMessage: string,
+  ): Promise<Response | never> {
+    if (response.ok) {
+      return response;
+    }
+
+    // An error downloading the content. Attempt to extract the reason behind it.
+    const text = await response.text();
+    let msg: string;
+    try {
+      const obj = JSON.parse(text);
+      msg =
+        obj.error || obj.message || obj.reason || JSON.stringify(obj, null, 2);
+    } catch {
+      msg = text;
+    }
+    throw new Error(`${errorMessage}.\n\nReason: ${msg}`);
+  }
+
+  private async readAndUnzip(
+    zipUrl: string,
+    unzipPath: string,
+    progress?: ProgressCallback,
+  ) {
+    const zipFile = Uri.parse(zipUrl).fsPath;
+    progress?.({
+      maxStep: this.LOG_DOWNLOAD_AND_PROCESS_PROGRESS_STEPS,
+      step: 2,
+      message: `Unzipping into ${basename(unzipPath)}`,
+    });
+    execFileSync("unzip", ["-q", "-d", unzipPath, zipFile]);
+  }
+
+  private async untargz(
+    tarballPath: string,
+    untarPath: string,
+    progress?: ProgressCallback,
+  ) {
+    progress?.({
+      maxStep: this.LOG_DOWNLOAD_AND_PROCESS_PROGRESS_STEPS,
+      step: 3,
+      message: `Untarring into ${basename(untarPath)}`,
+    });
+    void extLogger.log(`Untarring ${tarballPath} into ${untarPath}`);
+    execFileSync("tar", ["-xzf", tarballPath, "-C", untarPath]);
+  }
+
+  private async getPotentialTargetInfos(
+    experimentName: string,
+  ): Promise<Array<MinimalDownloadsType["targets"]["string"]>> {
+    const dir = mkdtempSync(
+      path.join(tmpDir.name, "codeql-compare-performance"),
+    );
+    const tasksDir = join(dir, "tasks");
+    await ensureDir(tasksDir);
+    // XXX hardcoded path
+    const dca = "/Users/esbena/Documents/codeql-dca/dca";
```

Comment:
Seems like this won't work for anyone else. Is there an easy way around this?

Reply:
good catch. I forgot about that hardcoding! Let me see what can be done.